### PR TITLE
clamp-js definition files added

### PR DESCRIPTION
**Pull request by Hikariii**
Please fill in this template.

- [X] Make your PR against the `master` branch.
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [X] Run `npm run lint -- package-name` if a `tslint.json` is present.

If adding a new definition:
- [X] The package does not provide its own types, and you can not add them.
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Run `tsc` without errors.
- [X] Include the required [files](https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package) and header. Base these on the README, *not* on an existing project.

### CODE REVIEW

**Comment by andy-ms** [comment]
Please use `npm run new-package -- clamp-js` to generate the correct package structure.

**Comment by DanielRosenwasser** [comment]
To elaborate on @andy-ms's feedback:

First, please add linting as other projects have done.

Second, use spaces over tabs, as the maintainers prefer it.

Lastly, can you change the style of your module to a UMD-style declaration file? It would look something like this:

Please change this to be

```ts
declare namespace clamp {
    export ClampOptions {
        // ...
    }
    // ...
}

function clamp(element: HTMLElement, options? clamp.ClampOptions): clamp.ClampResponse;

export = clamp;
export as namespace clamp;
```

**Comment by Hikariii** [comment]
Thank you for the suggestions and help, updated this to reflect correct structure and formatting

File: clamp-js/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "target": "es6",
+        "noImplicitAny": true,
+        "strictNullChecks": false,
```
Comment (andy-ms) [inline clamp-js/tsconfig.json]: This should be `true`, and add `"noImplicitThis": true,` above it.

**Pull request merged**